packages/components/src/date-time/types.ts

```diff
@@ -0,0 +1,72 @@
+export type DateTimeString = string;
+
+export type ValidDateTimeInput = Date | string | number | null;
+
+export interface TimePickerProps {
```

Comment:
All the `interface` declarations here could be switched to `type`.

<details>
<summary>Something like this might suffice</summary>

```diff
diff --git a/packages/components/src/date-time/types.ts b/packages/components/src/date-time/types.ts
index 3464f03aa0..ebd0312d28 100644
--- a/packages/components/src/date-time/types.ts
+++ b/packages/components/src/date-time/types.ts
@@ -2,7 +2,7 @@ export type DateTimeString = string;
 
 export type ValidDateTimeInput = Date | string | number | null;
 
-export interface TimePickerProps {
+export type TimePickerProps = {
 	/**
 	 * The initial current time the time picker should render.
 	 */
@@ -20,16 +20,16 @@ export interface TimePickerProps {
 	 * time as an argument.
 	 */
 	onChange?: ( time: DateTimeString ) => void;
-}
+};
 
-export interface DatePickerEvent {
+export type DatePickerEvent = {
 	/**
 	 * The date of the event.
 	 */
 	date: Date;
-}
+};
 
-export interface DatePickerProps {
+export type DatePickerProps = {
 	/**
 	 * The current date and time at initialization. Optionally pass in a `null`
 	 * value to specify no date is currently selected.
@@ -61,12 +61,13 @@ export interface DatePickerProps {
 	 * dot on the day of the event.
 	 */
 	events?: DatePickerEvent[];
-}
+};
 
-export interface DateTimePickerProps extends DatePickerProps, TimePickerProps {
-	/**
-	 * The function called when a new date or time has been selected. It is
-	 * passed the date and time as an argument.
-	 */
-	onChange?: ( date: DateTimeString | null ) => void;
-}
+export type DateTimePickerProps = DatePickerProps &
+	TimePickerProps & {
+		/**
+		 * The function called when a new date or time has been selected. It is
+		 * passed the date and time as an argument.
+		 */
+		onChange?: ( date: DateTimeString | null ) => void;
+	};

```
</details>